lib/web_ui/lib/src/engine/util.dart

```diff
@@ -560,3 +560,83 @@ String blurSigmasToCssString(double sigmaX, double sigmaY) {
 bool unsafeIsNull(dynamic object) {
   return object == null;
 }
+
+/// A typed variant of [html.Window.fetch].
+Future<html.Body> httpFetch(String url) async {
+  final dynamic result = await html.window.fetch(url);
+  return result as html.Body;
+}
+
+/// Extensions to [Map] that make it easier to treat it as a JSON object. The
+/// keys are `dynamic` because when JSON is deserialized from method channels
+/// it arrives as `Map<dynamic, dynamic>`.
+// TODO(yjbanov): use Json typedef when type aliases are shipped
+extension JsonExtensions on Map<dynamic, dynamic> {
```

Comment:
Is it generating same js code under O4 optimization?

Reply:
There's an overall increase of 0.03% (or 286 bytes) in code size for flutter_gallery when I compare `flutter build web --release` vs `flutter --local-engine=host_debug_unopt build web --release` with this PR. Some of these do become runtime casts (maybe something we can ask the Darat team to optimize for O4), but nearly all of these are not in the rendering pipeline. Most are I/O and platform messages.

Reply:
Thanks for checking. Suggestion: if you see anything worth optimizing here for rendering pipeline, you can use unsafeCast. Something like:
@pragma('dart2js:tryInline') // Required for next pragma to be effective.
@pragma('dart2js:as:trust') // Omits `as T`.
T unsafeCast<T>(dynamic any) => any as T;

Reply:
Interesting. We should try that pragma on the framework side for the most expensive `as` casts.